wiki/Client/File_formats/Osu_(file_format)/en.md

```diff
@@ -4,346 +4,623 @@
 
-### Holds (osu!mania only)
+`curve_points`
 
-*Hold syntax:* `x,y,time,type,hitSound,endTime:hitSample`
+> In the second example, the Bézier curve slider is composite. The control points of the first curve are (96,52), (200,200) and (250,200), and the control points of the second curve are (250,200) and (300,150). Duplicated control points denote the [red anchor point](/wiki/Hit_object/Slider_anchor).
 
-- **`endTime` (Integer):** End time of the hold, in milliseconds from the beginning of the beatmap's audio.
-- `x` determines the index of the column that the hold will be in. It is computed by `floor(x * columnCount / 512)` and clamped between `0` and `columnCount - 1`.
-- `y` does not affect holds. It defaults to the centre of the playfield, `192`.
+`slides`
 
-### Examples
+> The slider is returnable if the value is more than `1`.
+
+`length`
+
+> If the slider's `length` is longer than the defined curve, the slider will extend until it reaches the target length:
+>
+> - For bézier, catmull, and linear curves, it continues in a straight line from the end of the curve
+> - For perfect circle curves, it continues the circular arc
+>
+> The slider's `length` can be used to determine the time it takes to complete the slider. `length / (SliderMultiplier*100)*beatLength` tells how many milliseconds it takes to complete one slide of the slider (assuming `beatLength` has been adjusted for inherited timing points).
+
+`edge_sounds`
+
+> The first sound is the one that plays when the slider is first clicked, and the last sound is the one that plays when the slider's end is hit.
+>
+> Apart from `edge_sounds`, sliders also have an ongoing hitsound whenever the player is in range of the slider's follow circle. The sound file is looped for as long as it is active. This hitsound uses the hit object's `hitSound` and `hitSample` properties, but only the normal and whistle sounds are supported. Its filename is `<sampleSet>-slider<hitSound><index>.wav`, where `hitSound` is either `slide` for normal or `whistle` for whistle.
+>
+> Besides the sounds of the beginning and the end `edge_sounds`, sliders also have an ongoing hitsound whenever the player is in range of the slider's follow circle. The sound file is looped for as long as it is active. This hitsound uses the hit object's `hit_sound` and `hit_sample`, but only the normal and whistle sounds are supported. Its filename is `<sampleSet>-slider<hitSound><index>.wav`, where `hitSound` is either `slide` for normal or `whistle` for whistle.
+
+### Spinner
+
+#### Parameters
+
+`x,y,start_time,object_type,hit_sound,end_time,hit_sample`
+
+| Option | Value type | Description | Value |
+| :-- | :-- | :-- | :-- |
+| `x` | Integer | The location of the spinner on the X axis in pixels | `0` not used |
+| `y` | Integer | The location of the spinner on the Y axis in pixels | `0` not used |
```

Comment:
```suggestion
| `x` | Integer | *Unused* |  |
| `y` | Integer | *Unused* |  |
```

Reply:
What about this? I think we should explain what these parameters mean
| `x` | Integer | The location of the spinner on the X axis in pixels. Not used |
| `y` | Integer | The location of the spinner on the Y axis in pixels. Not used |